lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2912,11 +2931,20 @@ constructSharedTransaction
                 let refunds = case optionalDelegationAction of
                         Just Quit -> [W.stakeKeyDeposit pp]
                         _ -> []
+                delCerts <- case optionalDelegationAction of
+                    Nothing -> pure Nothing
+                    Just action -> do
+                        res <- liftHandler $ W.readSharedRewardAccount @n db wid
```

Comment:
At this point in the flow, we've already managed to create a balanced transaction, so I believe the `ErrConstructTxDelegationInvalid` would never been thrown from _here_.

Reply:
you are right. `ErrConstructTxDelegationInvalid` will never be thrown here. If ever, it would be caught earlier. removed that.